addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +113,117 @@ def _get_duration_from_tracking(self, trackings):
+    number of days before the resource is considered stale:
+        on the model pointed to by _track_duraton_field:
+            rotting_threshold_days = fields.Integer('Days to rot', default=0)
+
+    - The following attributes need to be overriden:
+      - _rotting_depends_fields array must be overriden to include the rotting_threshold_days field, as well as
+        any field that should affect the ability of a resource to rot (for example: a field that would mark a task as done):
+            _rotting_depends_fields = ['stage_id.rotting_threshold_days', 'won_status', 'type']
+
+      - is_rotting_domain must be overriden to include conditions taht must be met so that the field can be considered rotting:
+            is_rotting_domain = [
+                ('won_status', '=', 'pending'),
+                ('type', '=', 'opportunity')
+            ]
+
+        Note that the fields present in is_rotting_domain are the same as the ones present in _rotting_depends_fields
+
+    - The is_rotting, rotting_days fields need to be added to the relevant views
+        (as well as the "rotting_threshold_days" field on the tracked model).
+        You may want to use the rotting_form, rotting_kanban and rotting_list field widgets to display
+        the fields visually on form and kanban view
+        (please note- these widgets need both rotting_days and is_rotting fields on the view to function).
+
+    If the rotting_threshold_days field is not defined on the tracked module,
+    or if the value of rotting_threshold_days is 0,
+    then the resource will never rot.
+    """
+
+    # To override in inheriting models
+    _rotting_depends_fields = []
+    is_rotting_domain = [fields.Domain.TRUE]
+
+    rotting_days = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting', store=False)
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting', store=False)
+    rotting_date = fields.Date('Date at which the resource will start rotting', compute="_compute_rotting_date", store=True, readonly=False, required=False)
+
+    # Note for review:
+    # rotting_days_field is both a helper for writing the rotting filter,
+    # and a way to quickly check that a rotting field exists on the tracked model,
+    # as this was a pattern that was repeating a lot.
+    def rotting_days_field(self):
+        if 'rotting_threshold_days' in self[self._track_duration_field]:
+            return f'{self._track_duration_field}.rotting_threshold_days'
+        return False
+
+    def is_rotting_domain_base(self):
+        if self.rotting_days_field():
+            return [
+                ('rotting_date', '<=', fields.Datetime.today()),
+                (self.rotting_days_field(), '!=', 0),
+            ]
+        return [fields.Domain.FALSE]
+
+    @api.depends('write_date')
+    def _compute_rotting_date(self):
```

Comment:
Just to make sure because I have a slight doubt, "write_date" is written on when we create records correct?

To ensure we handle both create/write with this.